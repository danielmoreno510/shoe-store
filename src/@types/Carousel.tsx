type CarouselProps = {
  shoe: Shoe;
  width?: number;
};

type Shoe = {
  id: string;
  images: string[];
  reference: string;
  brand: string;
  sizes: Sizes[];
  price: number;
  size: number;
};

type Sizes = {
  label: string;
  value: string;
};

type ShoeImageProps = {
  item: string;
  size: number;
};
