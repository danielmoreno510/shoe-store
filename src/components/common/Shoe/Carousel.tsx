import React from 'react';
import {View, Image} from 'react-native';
import Carousel, {Pagination} from 'react-native-snap-carousel';

import styles from './styles';
import shoes from '../../../assets/images/shoes';
import {primaryColor} from '../../../constants';

export const ShoeImage: React.FC<ShoeImageProps> = ({item, size}) => (
  <Image style={[{width: size, height: size}]} source={shoes[item]} />
);

const CustomCarousel: React.FC<CarouselProps> = ({shoe, width = 180}) => {
  const [currentImage, setCurrentImage] = React.useState<number>(0);

  return (
    <View style={styles.carouselContainer}>
      <Carousel
        data={shoe.images}
        renderItem={({item}) => <ShoeImage item={item} size={width} />}
        sliderWidth={width}
        itemWidth={width}
        inactiveSlideOpacity={0.85}
        autoplayDelay={500}
        autoplayInterval={3000}
        onSnapToItem={index => setCurrentImage(index)}
        autoplay={true}
        loop
      />
      <Pagination
        dotsLength={shoe.images.length}
        activeDotIndex={currentImage}
        dotColor={primaryColor}
        inactiveDotColor="black"
        inactiveDotOpacity={0.4}
        inactiveDotScale={0.6}
      />
    </View>
  );
};

export default CustomCarousel;
