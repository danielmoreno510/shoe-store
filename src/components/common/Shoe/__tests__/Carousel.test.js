import React from 'react';
import {shallow} from 'enzyme';

import Carousel, {ShoeImage} from '../Carousel';

jest.spyOn(React, 'useState').mockImplementation(v => [v, jest.fn()]);

describe('Carousel component', () => {
  describe('GIVEN a set of valid props', () => {
    const testProps = {
      shoe: {images: []},
    };

    describe('WHEN the component is rendered', () => {
      const testPropsShoeImage = {item: {}, size: 20};
      const shallowWrapper = shallow(<Carousel {...testProps} />);
      const shallowSubWrapper = shallow(<ShoeImage {...testPropsShoeImage} />);
      shallowWrapper.find('mockConstructor').props().renderItem({item: {}});
      shallowWrapper.find('mockConstructor').props().onSnapToItem();

      it('THEN should display a regular Carousel', () => {
        expect(shallowWrapper).toMatchSnapshot();
        expect(shallowSubWrapper).toMatchSnapshot();
      });
    });
  });
});
