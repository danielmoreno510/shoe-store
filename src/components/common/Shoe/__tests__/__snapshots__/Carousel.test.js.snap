// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component GIVEN a set of valid props WHEN the component is rendered THEN should display a regular Carousel 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginBottom": -20,
    }
  }
>
  <mockConstructor
    autoplay={true}
    autoplayDelay={500}
    autoplayInterval={3000}
    data={Array []}
    inactiveSlideOpacity={0.85}
    itemWidth={180}
    loop={true}
    onSnapToItem={[Function]}
    renderItem={[Function]}
    sliderWidth={180}
  />
  <Component
    activeDotIndex={0}
    dotColor="#f1b30a"
    dotsLength={0}
    inactiveDotColor="black"
    inactiveDotOpacity={0.4}
    inactiveDotScale={0.6}
  />
</View>
`;

exports[`Carousel component GIVEN a set of valid props WHEN the component is rendered THEN should display a regular Carousel 2`] = `
<Image
  style={
    Array [
      Object {
        "height": 20,
        "width": 20,
      },
    ]
  }
/>
`;
